src/domain/email/entities/email.entity.ts

```diff
@@ -0,0 +1,17 @@
+export class InvalidEmailFormatError extends Error {
+  constructor() {
+    super('Provided email is not a recognizable email format.');
+  }
+}
+
+export class Email {
+  // https://www.ietf.org/rfc/rfc5322.txt
+  private static EMAIL_REGEX: RegExp =
```

Comment:
Can we consider this to a followup PR? I'd need to reevaluate on how to integrate AJV with this since right now this validation is executed within the Email entity.

Reply:
Sure! No problem on my side! 🙂 And as I said, I don't have a strong opinion regarding AJV versus a custom validation within the entity 👍🏻

Reply:
> Sure! No problem on my side! 🙂 And as I said, I don't have a strong opinion regarding AJV versus a custom validation within the entity 👍🏻

Sounds good. Let's see how the entities in the domain evolve with this feature then 👍